packages/cli/src/validateNonInterActiveAuth.ts

```diff
@@ -14,15 +14,17 @@ export async function validateNonInteractiveAuth(
 ) {
   const effectiveAuthType =
     configuredAuthType ||
-    (process.env.GOOGLE_GENAI_USE_VERTEXAI === 'true'
-      ? AuthType.USE_VERTEX_AI
-      : process.env.GEMINI_API_KEY
-        ? AuthType.USE_GEMINI
-        : undefined);
+    (process.env.GOOGLE_GENAI_USE_GCP === 'true'
```

Comment:
nit: this conditional is getting a bit hard to read. as a fast follow can you refactor it to use a switch by AuthType or other technique to make it not as nested.